## Extracción de Datos - API Deezer

#### Librerías

In [22]:
import requests
import time
import pandas as pd
import json


#### Lista de 30 artistas populares (Argentina, España y EEUU)

In [2]:
lista_artistas=[
 
    # 🇺🇸 EEUU (10)
    "Taylor Swift", "Drake", "Morgan Wallen", "Kendrick Lamar", "The Weeknd",
    "SZA", "Zach Bryan", "Tyler the Creator", "Billie Eilish", "Ariana Grande",
 
    # 🇪🇸 España (10)
    "Bad Bunny", "Karol G", "Myke Towers", "Quevedo", "Aitana",
    "Rosalía", "Morad", "Melendi", "Enrique Iglesias", "Juanes",
 
    # 🇦🇷 Argentina (10)
    "Duki", "María Becerra", "Tini Stoessel", "Nicki Nicole", "Cazzu",
    "Lali", "Paulo Londra", "Shakira", "Bizarrap", "Emilia",
]
 
print(f"Total: {len(lista_artistas)} artistas ✅")

Total: 30 artistas ✅


#### Consulta de artistas en Deezer API - se limita a 1 por existir duplicidad de ID's - RUTA:SEARCH

In [3]:
lista_artistas=[
 
    # 🇺🇸 EEUU (10)
    "Taylor Swift", "Drake", "Morgan Wallen", "Kendrick Lamar", "The Weeknd",
    "SZA", "Zach Bryan", "Tyler the Creator", "Billie Eilish", "Ariana Grande",
 
    # 🇪🇸 España (10)
    "Bad Bunny", "Karol G", "Myke Towers", "Quevedo", "Aitana",
    "Rosalía", "Morad", "Melendi", "Enrique Iglesias", "Juanes",
 
    # 🇦🇷 Argentina (10)
    "Duki", "María Becerra", "Tini Stoessel", "Nicki Nicole", "Cazzu",
    "Lali", "Paulo Londra", "Shakira", "Bizarrap", "Emilia",
]

artista_search=[]

for artista in lista_artistas:
    consulta=requests.get(f"https://api.deezer.com/search/artist?q={artista}&index=0&limit=1")
    consulta.status_code

    if consulta.status_code != 200:
        print(f"No se pudo realizar la consulta del artista: {artista}.\n Error : {consulta.status_code}") 

    datos=consulta.json()
    artista_search.append(datos['data'][0])

print(artista_search)



[{'id': 12246, 'name': 'Taylor Swift', 'link': 'https://www.deezer.com/artist/12246', 'picture': 'https://api.deezer.com/artist/12246/image', 'picture_small': 'https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/56x56-000000-80-0-0.jpg', 'picture_medium': 'https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/250x250-000000-80-0-0.jpg', 'picture_big': 'https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/500x500-000000-80-0-0.jpg', 'picture_xl': 'https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/1000x1000-000000-80-0-0.jpg', 'nb_album': 124, 'nb_fan': 12524357, 'radio': True, 'tracklist': 'https://api.deezer.com/artist/12246/top?limit=50', 'type': 'artist'}, {'id': 246791, 'name': 'Drake', 'link': 'https://www.deezer.com/artist/246791', 'picture': 'https://api.deezer.com/artist/246791/image', 'picture_small': 'https://cdn-images.dzcdn.net/images/artist/5d2fa7f140a6bdc2c864c3465a61fc71/56x56-0

In [4]:
len(artista_search)

30

In [5]:
import json
print(json.dumps(artista_search, indent=2, ensure_ascii=False))     # para ver estructura

[
  {
    "id": 12246,
    "name": "Taylor Swift",
    "link": "https://www.deezer.com/artist/12246",
    "picture": "https://api.deezer.com/artist/12246/image",
    "picture_small": "https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/56x56-000000-80-0-0.jpg",
    "picture_medium": "https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/250x250-000000-80-0-0.jpg",
    "picture_big": "https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/500x500-000000-80-0-0.jpg",
    "picture_xl": "https://cdn-images.dzcdn.net/images/artist/e528e270424103b527f8a27ac625563b/1000x1000-000000-80-0-0.jpg",
    "nb_album": 124,
    "nb_fan": 12524357,
    "radio": true,
    "tracklist": "https://api.deezer.com/artist/12246/top?limit=50",
    "type": "artist"
  },
  {
    "id": 246791,
    "name": "Drake",
    "link": "https://www.deezer.com/artist/246791",
    "picture": "https://api.deezer.com/artist/246791/image",
    "picture_small": "h

#### Subsanación de Listado de ID's

In [47]:
lista_id=[]
for d in artista_search:
    lista_id.append(d['id'])
    lista_id.append(d['name'])
print(lista_id)
    

[12246, 'Taylor Swift', 246791, 'Drake', 386950521, 'Morgan Wallen', 525046, 'Kendrick Lamar', 4050205, 'The Weeknd', 5531258, 'SZA', 71855892, 'Zach Bryan', 1194083, 'Tyler, The Creator', 9635624, 'Billie Eilish', 1562681, 'Ariana Grande', 10583405, 'Bad Bunny', 5297021, 'KAROL G', 12029862, 'Myke Towers', 6705223, 'Quevedo', 11928391, 'Aitana', 554792, 'ROSALÍA', 111130212, 'Morad', 2697, 'Melendi', 2792, 'Enrique Iglesias', 70, 'Juanes', 4902904, 'DUKI', 14343187, 'Maria Becerra', 2252591, 'TINI', 13299379, 'Nicki Nicole', 12268072, 'Cazzu', 5181388, 'Lali', 12637039, 'Paulo Londra', 267400032, 'Shakira', 12487862, 'Bizarrap', 61094422, 'Emilia']


##### Se observan ID incorrectos en la API debido a la generación de nuevos perfiles (la API permite adeicionalmente las acciones de PUSH/DELETE).
##### Se corrigen los ID de Shakira '267400032' se reemplaza por '160' y ID  de Morgan Wallen '386950521' por '7188840'.

In [102]:
artistas_info = []
for artista in artista_search:
    artistas_info.append({"id_artista": artista["id"], "nombre_artista":artista["name"]})
    for artista in artistas_info:     
        if artista["nombre_artista"] == "Morgan Wallen":         
            artista["id_artista"] = 7188840
        if artista["nombre_artista"] == "Shakira":         
            artista["id_artista"] = 160

artistas_info

[{'id_artista': 12246, 'nombre_artista': 'Taylor Swift'},
 {'id_artista': 246791, 'nombre_artista': 'Drake'},
 {'id_artista': 7188840, 'nombre_artista': 'Morgan Wallen'},
 {'id_artista': 525046, 'nombre_artista': 'Kendrick Lamar'},
 {'id_artista': 4050205, 'nombre_artista': 'The Weeknd'},
 {'id_artista': 5531258, 'nombre_artista': 'SZA'},
 {'id_artista': 71855892, 'nombre_artista': 'Zach Bryan'},
 {'id_artista': 1194083, 'nombre_artista': 'Tyler, The Creator'},
 {'id_artista': 9635624, 'nombre_artista': 'Billie Eilish'},
 {'id_artista': 1562681, 'nombre_artista': 'Ariana Grande'},
 {'id_artista': 10583405, 'nombre_artista': 'Bad Bunny'},
 {'id_artista': 5297021, 'nombre_artista': 'KAROL G'},
 {'id_artista': 12029862, 'nombre_artista': 'Myke Towers'},
 {'id_artista': 6705223, 'nombre_artista': 'Quevedo'},
 {'id_artista': 11928391, 'nombre_artista': 'Aitana'},
 {'id_artista': 554792, 'nombre_artista': 'ROSALÍA'},
 {'id_artista': 111130212, 'nombre_artista': 'Morad'},
 {'id_artista': 2697

### Consulta de tracks (2 resultados incompletos Taylor S y TINI)

In [23]:
import requests
import time

def consulta_tracks(limit_por_artista, delay):
    lista_artistas = [
        "Taylor Swift", "Drake", "Morgan Wallen", "Kendrick Lamar", "The Weeknd",
        "SZA", "Zach Bryan", "Tyler the Creator", "Billie Eilish", "Ariana Grande",
        "Bad Bunny", "Karol G", "Myke Towers", "Quevedo", "Aitana",
        "Rosalía", "Morad", "Melendi", "Enrique Iglesias", "Juanes",
        "Duki", "María Becerra", "Tini Stoessel", "Nicki Nicole", "Cazzu",
        "Esposito", "Paulo Londra", "Shakira", "Bizarrap", "Emilia",
    ]
    
    #headers para evitar bloqueos
    headers = {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
    }
    
    tracks_search = {'id': [], 'title': [], 'rank': [], 'artist': []}
    
    for artista in lista_artistas:
        url= f"https://api.deezer.com/search/track?q=artist:\"{artista}\"&index=1&limit={limit_por_artista}"
        #url_1= f"https://api.deezer.com/search/track?q=artist:\"{artista}\"&limit={limit_por_artista}"
        print(f"Consultando tracks para: {artista}")
        
        try:
            respuesta = requests.get(url, headers=headers)
            if respuesta.status_code != 200:
                print(f"Error {respuesta.status_code} para {artista}")
                continue
            
            datos = respuesta.json()
            tracks = datos.get('data', [])
            
            if not tracks:              #en primer consulta arrojó 2 canciones para Taylor S.
                print(f"No se encontraron tracks para {artista}")
                continue
            
            for track in tracks:            #ver de agregar variables
                tracks_search['id'].append(track['id'])
                tracks_search['title'].append(track['title'])
                tracks_search['rank'].append(track.get('rank', 0))
                tracks_search['artist'].append(artista)  # guardamos el nombre buscado
            
            print(f"  → {len(tracks)} tracks obtenidos de {artista}")
            
        except Exception as e:
            print(f"Excepción con {artista}: {e}")
        
        # tiempo de espera para no sobrecargar la API
        time.sleep(delay)
    
    return tracks_search

# llamado y conversión a tabla
resultado = consulta_tracks(limit_por_artista=100, delay=1)
print(f"Total de tracks recolectados: {len(resultado['id'])}")

import pandas as pd
df = pd.DataFrame(resultado)
print(df.head())



Consultando tracks para: Taylor Swift
Excepción con Taylor Swift: HTTPSConnectionPool(host='api.deezer.com', port=443): Max retries exceeded with url: /search/track?q=artist:%22Taylor%20Swift%22&index=1&limit=100 (Caused by NameResolutionError("HTTPSConnection(host='api.deezer.com', port=443): Failed to resolve 'api.deezer.com' ([Errno -2] Name or service not known)"))
Consultando tracks para: Drake
Excepción con Drake: HTTPSConnectionPool(host='api.deezer.com', port=443): Max retries exceeded with url: /search/track?q=artist:%22Drake%22&index=1&limit=100 (Caused by NameResolutionError("HTTPSConnection(host='api.deezer.com', port=443): Failed to resolve 'api.deezer.com' ([Errno -2] Name or service not known)"))
Consultando tracks para: Morgan Wallen
Excepción con Morgan Wallen: HTTPSConnectionPool(host='api.deezer.com', port=443): Max retries exceeded with url: /search/track?q=artist:%22Morgan%20Wallen%22&index=1&limit=100 (Caused by NameResolutionError("HTTPSConnection(host='api.deeze

In [24]:
#print(resultado)
print(json.dumps(resultado, indent=2, ensure_ascii=False)) #para ver estructura (diccionario con listas por cada columna)

{
  "id": [
    446082632,
    3106586641,
    3106586681,
    67929157,
    350171311,
    350171251,
    3106586631,
    2715581062,
    1476644212,
    350171301,
    350171291,
    1766771507,
    3152680431,
    67929154,
    511062642,
    72089437,
    1766771527,
    350171331,
    2775438091,
    1766771497,
    372049961,
    456011702,
    52735971,
    1766771557,
    134533240,
    723393562,
    370426011,
    1766771587,
    97206068,
    3106586701,
    3106586651,
    77511988,
    3106586721,
    67929160,
    97206060,
    723393442,
    3106586671,
    623422582,
    634438752,
    723393612,
    723393582,
    67929167,
    350171271,
    67929162,
    1987477707,
    1766771607,
    723393482,
    67929163,
    67929161,
    350171341,
    1766771627,
    2783963112,
    1048706692,
    136889416,
    120380320,
    1766771677,
    75072823,
    3106586711,
    67929155,
    67929165,
    3106586691,
    723393422,
    131391770,
    3855225991,
    97206008,
    

In [25]:
df=pd.DataFrame(resultado)
df

,id,title,rank,artist
0,446082632,"All The Stars (From ""Black Panther: The Album"")",927928,Kendrick Lamar
1,3106586641,luther,824685,Kendrick Lamar
2,3106586681,tv off,827487,Kendrick Lamar
3,67929157,Money Trees,823308,Kendrick Lamar
4,350171311,HUMBLE.,816596,Kendrick Lamar
...,...,...,...,...
2563,1551939962,Te Doy Mi Vida (Masters en Parranda),240089,Emilia
2564,894202362,HISTERIQUEO,203814,Emilia
2565,831895912,Billion,271307,Emilia
2566,2498034101,A_1000_Km.mp3,198967,Emilia


#### Consulta modelo de tracks (Taylor Swift y Tini Stoessel - 100 canciones)

In [26]:
import re

def llamar():
    #url='https://api.deezer.com/search/track?q=taylor&swift=undefined&index=0&limit=100'    """FUNCION OK PARA TAYLOR"""
    url00="https://api.deezer.com/search/track?q=TINI&index=0&limit=100" #"id": "2252591","name": "TINI",
    headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
    }


    respuesta = requests.get(url00,headers=headers)
    if respuesta.status_code != 200:
        print(f"Error: {respuesta.status_code}")
    else:
        datos = respuesta.json()
        tracks = datos.get('data',[])
        return tracks
      
        

          



In [27]:
consulta_tini=llamar()
#print(json.dumps(a,indent=2,ensure_ascii=False))



In [28]:
len(consulta_tini)

96

In [29]:
print(json.dumps(consulta_tini,indent=2,ensure_ascii=False))

[
  {
    "id": 3288087861,
    "readable": true,
    "title": "blackout ?",
    "title_short": "blackout ?",
    "title_version": "",
    "isrc": "USSD12500117",
    "link": "https://www.deezer.com/track/3288087861",
    "duration": 180,
    "rank": 577836,
    "explicit_lyrics": false,
    "explicit_content_lyrics": 0,
    "explicit_content_cover": 2,
    "preview": "https://cdnt-preview.dzcdn.net/api/1/1/4/9/3/0/49390d5f9b4a2ab72f844858653c663e.mp3?hdnea=exp=1777380704~acl=/api/1/1/4/9/3/0/49390d5f9b4a2ab72f844858653c663e.mp3*~data=user_id=0,application_id=42~hmac=cf0d2e3eb9a6099a3f9f3ea3100a7fa49e5d29ff1823cc00344feffafaa7021e",
    "md5_image": "fee888110d29c7d8021275c36004961f",
    "artist": {
      "id": 61094422,
      "name": "Emilia",
      "link": "https://www.deezer.com/artist/61094422",
      "picture": "https://api.deezer.com/artist/61094422/image",
      "picture_small": "https://cdn-images.dzcdn.net/images/artist/bd2b4dd505269891296688261244d10e/56x56-000000-80-0-0.jpg

##### Función para limpiar tracks

In [30]:
def depurar():  
    lista_tini_depurada=[]
    for track in consulta_tini:
        if "TINI" not in track['artist']['name']:
            continue
        else:
            lista_tini_depurada.append(track)
            
    print(f"Se encontraron {len(lista_tini_depurada)} resultados.")
            
    
    return lista_tini_depurada
TINI=depurar()

Se encontraron 72 resultados.


In [31]:
for r in TINI:
    print(f" - {r['title']}")

#SE OBSERVAN RESULTADOS DUPLICADOS 

 - Miénteme
 - Dos Amantes
 - Universidad
 - Consejo de Amor (feat. Morat)
 - Fantasi
 - el cielo
 - Maldita Foto
 - Fresa
 - Un Beso en Madrid
 - Una Noche Más
 - pa
 - El Cielo + Fresa LIVE
 - Te Pido Live
 - Cupido
 - Bar
 - La Loto
 - Carne y Hueso
 - Muñecas
 - tinta 90
 - Te Quiero Más
 - Las Jordans
 - Te Pido
 - Oye
 - La Triple T
 - Missed Calls
 - 36 Vidas
 - Suéltate el Pelo
 - Cupido LIVE
 - 22
 - Siempre Brillarás
 - me voy en vivo
 - Por Qué Te Vas
 - Down
 - ni de ti
 - 7 Veces
 - Princesa
 - Podemos
 - ángel
 - Ser Mejor
 - Acércate
 - Si Tu Te Vas
 - buenos aires
 - What If, Then What? (feat. Amiture)
 - Quiero Volver
 - El Último Beso
 - De Papel
 - En Mi Mundo (From "Violetta" Music from the TV Series)
 - No Reply
 - Libre Soy (Single Version)
 - Te Creo
 - Una Noche Más LIVE
 - Confía En Mí
 - Yo Te Amo A Ti
 - me voy
 - Te Creo LIVE
 - Esto No Puede Terminar
 - buenos aires en vivo
 - Recuerdo
 - Te Olvidaré
 - Ya No Hay Nadie Que Nos Pare
 - Hoy Somos Más (Live)
 

## Consulta de canciones por Id artista (RUTA:TOP)

In [17]:
import requests
import time

def consulta_cancion(lista_artistas):
    info_canciones = []
       
    for artista in lista_artistas:
        id_artista = artista["id_artista"]
        nombre_artista = artista["nombre_artista"]

        consulta_canciones = requests.get(f"https://api.deezer.com/artist/{id_artista}/top?limit=50")

        if consulta_canciones.status_code != 200:
            print(f"No se pudo consultar las canciones del artista: {nombre_artista}")
            continue
        datos_canciones = consulta_canciones.json()

        for cancion in datos_canciones["data"]:
            info_canciones.append({"id_artista": id_artista,"nombre_artista": nombre_artista,"id_cancion": cancion["id"],
                            "titulo_cancion": cancion["title"],"id_album":cancion ["album"]["id"]})#,"titulo_album": cancion["album"]["title"]})#,
                            #"contributors": cancion.get("contributors")})

            
        time.sleep(0.3)
    return info_canciones


In [18]:
artistas_30= [{'id_artista': 12246, 'nombre_artista': 'Taylor Swift'},
 {'id_artista': 246791, 'nombre_artista': 'Drake'},
 {'id_artista': 7188840, 'nombre_artista': 'Morgan Wallen'},
 {'id_artista': 525046, 'nombre_artista': 'Kendrick Lamar'},
 {'id_artista': 4050205, 'nombre_artista': 'The Weeknd'},
 {'id_artista': 5531258, 'nombre_artista': 'SZA'},
 {'id_artista': 71855892, 'nombre_artista': 'Zach Bryan'},
 {'id_artista': 1194083, 'nombre_artista': 'Tyler, The Creator'},
 {'id_artista': 9635624, 'nombre_artista': 'Billie Eilish'},
 {'id_artista': 1562681, 'nombre_artista': 'Ariana Grande'},
 {'id_artista': 10583405, 'nombre_artista': 'Bad Bunny'},
 {'id_artista': 5297021, 'nombre_artista': 'KAROL G'},
 {'id_artista': 12029862, 'nombre_artista': 'Myke Towers'},
 {'id_artista': 6705223, 'nombre_artista': 'Quevedo'},
 {'id_artista': 11928391, 'nombre_artista': 'Aitana'},
 {'id_artista': 554792, 'nombre_artista': 'ROSALÍA'},
 {'id_artista': 111130212, 'nombre_artista': 'Morad'},
 {'id_artista': 2697, 'nombre_artista': 'Melendi'},
 {'id_artista': 2792, 'nombre_artista': 'Enrique Iglesias'},
 {'id_artista': 70, 'nombre_artista': 'Juanes'},
 {'id_artista': 4902904, 'nombre_artista': 'DUKI'},
 {'id_artista': 14343187, 'nombre_artista': 'Maria Becerra'},
 {'id_artista': 2252591, 'nombre_artista': 'TINI'},
 {'id_artista': 13299379, 'nombre_artista': 'Nicki Nicole'},
 {'id_artista': 12268072, 'nombre_artista': 'Cazzu'},
 {'id_artista': 5181388, 'nombre_artista': 'Lali'},
 {'id_artista': 12637039, 'nombre_artista': 'Paulo Londra'},
 {'id_artista': 160, 'nombre_artista': 'Shakira'},
 {'id_artista': 12487862, 'nombre_artista': 'Bizarrap'},
 {'id_artista': 61094422, 'nombre_artista': 'Emilia'}]

info_artista_cancion=consulta_cancion(artistas_30)


In [19]:
print(info_artista_cancion)

[{'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685431, 'titulo_cancion': 'The Fate of Ophelia', 'id_album': 829966251}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685451, 'titulo_cancion': 'Opalite', 'id_album': 829966251}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 89077547, 'titulo_cancion': 'Blank Space', 'id_album': 9007781}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 89077555, 'titulo_cancion': 'Shake It Off', 'id_album': 9007781}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 2512293211, 'titulo_cancion': "Out Of The Woods (Taylor's Version)", 'id_album': 504180521}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 737967292, 'titulo_cancion': 'Cruel Summer', 'id_album': 108447472}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685441, 'titulo_cancion': 'Elizabeth Taylor', 'id_album': 829966251}, 

In [23]:
df=pd.DataFrame(info_artista_cancion)

In [24]:
df.head(5)

,id_artista,nombre_artista,id_cancion,titulo_cancion,id_album
0,12246,Taylor Swift,3579685431,The Fate of Ophelia,829966251
1,12246,Taylor Swift,3579685451,Opalite,829966251
2,12246,Taylor Swift,89077547,Blank Space,9007781
3,12246,Taylor Swift,89077555,Shake It Off,9007781
4,12246,Taylor Swift,2512293211,Out Of The Woods (Taylor's Version),504180521


## Consulta de album por Id Canción (RUTA: ALBUM)

In [ ]:
def consulta_album(info_cancion):
    id_canciones = []
    info_album= []
    for cancion in info_cancion:
        id_cancion = cancion["id_cancion"]
        titulo_cancion = cancion["titulo_cancion"]
        id_album = cancion["id_album"]

        consulta_album = requests.get(f"https://api.deezer.com/album/{id_album}")

        if consulta_album.status_code != 200:
            print(f"No se pudo consultar la canción {id_cancion}: {titulo_cancion}.")
            continue

        datos_album = consulta_album.json()

        info_album.append({'id_cancion':{id_cancion}, 'titulo_cancion':{titulo_cancion}, 'id_album':{id_album},
                          'id_genre':datos_album['id_genre'],
                          'genre': datos_album['genres']['data']['name'], 
                          'año_de_lanzamiento':datos_album['release_date']})
    return info_album
    
    

In [46]:
canciones_a_buscar=info_artista_cancion
print(canciones_a_buscar)


[{'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685431, 'titulo_cancion': 'The Fate of Ophelia', 'id_album': 829966251}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685451, 'titulo_cancion': 'Opalite', 'id_album': 829966251}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 89077547, 'titulo_cancion': 'Blank Space', 'id_album': 9007781}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 89077555, 'titulo_cancion': 'Shake It Off', 'id_album': 9007781}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 2512293211, 'titulo_cancion': "Out Of The Woods (Taylor's Version)", 'id_album': 504180521}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 737967292, 'titulo_cancion': 'Cruel Summer', 'id_album': 108447472}, {'id_artista': 12246, 'nombre_artista': 'Taylor Swift', 'id_cancion': 3579685441, 'titulo_cancion': 'Elizabeth Taylor', 'id_album': 829966251}, 

In [47]:
resultado=consulta_album(canciones_a_buscar)

KeyError: 'rank'

ID del artista
Nombre del artista
Título de la canción
Título del álbum

Tipo (colaboración, single, etc.)
Año de lanzamiento
Género
Id del género